### ISSUE TITLE
Reference style markdown links can pull from other pages

### ISSUE THREAD

**Issue by parlough** [simolus3/built_site#6]
For example, if `issue-1.md` has a reference link like:

```md
This is my [link][].

[link]: https://first-link.example
```

and `issue-2.md` has the same reference like:

```md
Make sure to check out [link][].

[link]: https://other-link.example
```

Then the generated HTML for `issue-2` will link to the destination defined in `issue-1`:

```
<a href="https://first-link.example">link</a>
```

Personally, I expected reference links to only work on the current page, but perhaps I'm misunderstanding something or making another mistake.

**Comment by simolus3** [comment]
Good catch, this must have been horrible to diagnose...

Fixed in `built_site: 0.2.12`.

**Comment by parlough** [comment]
I was a bit confused at first, that's for sure!

Thanks for fixing this :)
